src/main/java/startsteps/ECommerceShop/service/CartService.java

```diff
@@ -98,4 +96,29 @@ private List<CartProductResponse> createCartProduct(List<CartProduct> cartProduc
                 cartProduct.getQuantity(),
                 cartProduct.getPrice())).collect(Collectors.toList());
         }
+
+    @Transactional
+    public CartResponse getCart(User user){
+        Cart cart = user.getCart();
+
+        if (cart == null || cart.getCartProductList().isEmpty()){
+            throw new EmptyCartException("You haven't added anything yet, your cart is empty");
+        }
+        log.info("get cart: {}", user.getCart());
+
+        List<CartProductResponse> cartProductResponses = cart.getCartProductList().stream()
+                .map(this::mapToCartProductResponse).collect(Collectors.toList());
+
+        return new CartResponse("Cart details retrieved successfully", cartProductResponses, cart.getTotalPrice());
+    }
+
+    private CartProductResponse mapToCartProductResponse(CartProduct cartProduct){
```

Comment:
you can have a constructor in CartProductResponse that takes in one argument of Product and construct the object that way

Reply:
done, implemented a constructor.

Reply:
you can delete this function now

Reply:
done